### PR TITLE
feat: add Dockerfile

### PR DESCRIPTION
**Pull request by lmilbaum**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+RUN install-tool node v18.17.0
+
+# renovate: datasource=npm versioning=npm
+RUN install-tool yarn 1.22.19
+
+# renovate: datasource=github-releases packageName=moby/moby
+RUN install-tool docker v24.0.5
+
+# Numeric user ID for the ubuntu user. Used to indicate a non-root user to OpenShift
+USER 1000
```
Comment (viceice) [inline Dockerfile:21]: trailing newline

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,21 @@
+FROM ghcr.io/containerbase/base:9.8.2@sha256:f41dc51323ab4839fa7e1242750363497fa45aec9d08f93ce2b42c5b10a581e7 AS base
```
Comment (viceice) [inline Dockerfile:1]: needs to be multi stage to build full image too. should depend on build arg.

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -2,18 +2,40 @@ name: build
...
+        uses: docker/metadata-action@v4.6.0
+        with:
+          images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
 
-      - name: Run a multi-line script
-        run: |
-          echo Add other actions to buil,
-          echo test, and deploy your project.
+      - name: Build and push Docker image
+        uses: docker/build-push-action@v4.1.1
```
Comment (viceice) [inline .github/workflows/build.yml:36]: no, we will use semantic release to publish images, like we do on containerbase.

**Review by viceice — COMMENTED** [review]
please add prettier and editor config first

File: Dockerfile (hunk 2 above)
Comment (lmilbaum) [reply Dockerfile:1]: Can we defer this change to a following PR?

File: .github/workflows/build.yml (hunk 3 above)
Comment (lmilbaum) [reply .github/workflows/build.yml:36]: Agreed. Can we defer this change to a later stage?

File: Dockerfile (hunk 2 above)
Comment (viceice) [reply Dockerfile:1]: ok, but disable push to registry until we've proper image configurations.

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -2,18 +2,40 @@ name: build
...
+        with:
+          context: .
+          push: ${{ github.event_name != 'pull_request' }}
```
Comment (viceice) [inline .github/workflows/build.yml:39]:
```suggestion
          push: false
```
no push for now

File: .github/workflows/build.yml [hunk 5]
```diff
@@ -2,18 +2,40 @@ name: build
...
+      contents: read
+      packages: none
+
     steps:
-      - uses: actions/checkout@v3
+      - name: Checkout repository
+        uses: actions/checkout@v3.5.3
+
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v2.2.0
```
Comment (viceice) [inline .github/workflows/build.yml:23]: only on main, will fail on PR

File: Dockerfile (hunk 1 above)
Comment (viceice) [inline Dockerfile:21]: remove, this will be done on renovate images

File: Dockerfile [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+  org.opencontainers.image.licenses="AGPL-3.0-only"
+
+# prepare all tools
+RUN prepare-tool all
+
+# renovate: datasource=node
+RUN install-tool node v18.17.0
+
+# renovate: datasource=npm versioning=npm
+RUN install-tool yarn 1.22.19
```
Comment (viceice) [inline Dockerfile:15]: we should now use corepack instead. it will provide yarn

**Review by viceice — APPROVED** [review]

**Pull request merged**